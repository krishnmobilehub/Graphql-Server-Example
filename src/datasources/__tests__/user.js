const UserAPI = require('../user');

const mockStore = {
    users: {
      find: jest.fn(),
      create: jest.fn(),
    },
  };
  module.exports.mockStore = mockStore;

const ds = new UserAPI({ store: mockStore });
ds.initialize({ context: { user: { id: 1, username: 'test', password:'test@123', phoneNumber:'9876543210',verificationCode:'2344' } } });

describe('[UserAPI.findUserByUserName]', () => {
    it('returns null for invalid username', async () => {
      const res = await ds.findUserByUserName("test");
      expect(res).toMatchSnapshot();
    });  
});

describe('[UserAPI.getUserById]', () => {
    it('looks up user by userId', async () => {
      const res = await ds.getUserById("1");
      expect(res).toMatchSnapshot();
    });
  
    it('returns empty array if nothing found', async () => {
      const res = await ds.getUserById("1");
      expect(res).toMatchSnapshot();
    });
  });