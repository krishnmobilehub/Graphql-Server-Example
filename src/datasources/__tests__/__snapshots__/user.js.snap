// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UserAPI.findUserByUserName] returns null for invalid username 1`] = `null`;

exports[`[UserAPI.getUserById] looks up user by userId 1`] = `null`;

exports[`[UserAPI.getUserById] returns empty array if nothing found 1`] = `null`;
